src/domain/interfaces/auth-api.interface.ts

```diff
@@ -9,4 +9,10 @@ export interface IAuthApi {
     message: SiweMessage;
     signature: `0x${string}`;
   }): Promise<boolean>;
+
+  cacheNonce(nonce: string): Promise<void>;
+
+  getCachedNonce(nonce: string): Promise<string | undefined>;
```

Comment:
Similar to the above ☝️ 

```suggestion
  getNonce(nonce: string): Promise<string | undefined>;
```

Reply:
Changed in b15fa98e39c07d2987f9f21eafce4ea6e318ec21.